### PR TITLE
apply copy to clipboard feature for copying Proposal id & SEP code

### PR DESCRIPTION
**Pull request by Junjiequan**
## Description

1. As a user I want to be able to copy IDs easily by clicking the helper "Copy to clipboard"  button.
2. E2e test for copy clipboard is covered
3. `cypress-real-events` package is installed and used to simulate real events
4. e2e test fail fix also included - ```Timed out retrying after 15000ms: Expected to find element: `[aria-label="Edit"]`, but never found it.``` error pops up sometimes.  Replaced `.parent()` method with `.closest()` to get the element more effectively.

 

e2e test 

## How Has This Been Tested

* manual test
* e2e test

## Fixes
Dec-08-2022 15-52-15

## Changes

apps/user-office-frontend/src/components/proposal/ProposalTable.tsx
apps/user-office-frontend/src/components/proposal/ProposalTableOfficer.tsx
apps/user-office-frontend/src/components/SEP/Proposals/SEPProposalsAndAssignmentsTable.tsx
apps/user-office-frontend/src/components/SEP/SEPsTable.tsx

apps/user-office-frontend-e2e/cypress/integration/generalSeps.ts
apps/user-office-frontend-e2e/cypress/integration/proposals.ts 
apps/user-office-frontend-e2e/cypress/integration/SEP.ts 
apps/user-office-frontend-e2e/cypress/support/utils.ts
apps/user-office-frontend-e2e/package.json

## Tests included/Docs Updated?

- [x] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: apps/user-office-frontend-e2e/cypress/integration/proposals.ts [hunk 1]
```diff
@@ -120,6 +120,21 @@ context('Proposal tests', () => {
       });
     });
 
+    it('Copy to clipboard should work for Proposal ID ', () => {
```
Comment (martin-trajanovski) [inline apps/user-office-frontend-e2e/cypress/integration/proposals.ts:123]: Nothing special but I just noticed a small detail here. There is one blank space in the end of the test title that could be removed.

File: apps/user-office-frontend/src/components/SEP/Proposals/SEPProposalsAndAssignmentsTable.tsx [hunk 2]
```diff
@@ -85,7 +98,7 @@ const SEPProposalColumns = [
   },
   {
     title: 'Average grade',
-    render: (rowData: SEPProposalType): string => {
+    render: (rowData) => {
```
Comment (martin-trajanovski) [inline apps/user-office-frontend/src/components/SEP/Proposals/SEPProposalsAndAssignmentsTable.tsx:101]: Nice cleanup in this file 👍

**Review by martin-trajanovski — APPROVED** [review]
This looks great to me. I just left a small comment about removing a blank space but otherwise nice job 💯

File: apps/user-office-frontend/src/components/SEP/Proposals/SEPProposalsAndAssignmentsTable.tsx [hunk 3]
```diff
@@ -114,7 +127,7 @@ const SEPProposalColumns = [
 
       return isNaN(stdDeviation) ? '-' : `${stdDeviation}`;
     },
-    customSort: (a: SEPProposalType, b: SEPProposalType) =>
+    customSort: (a, b) =>
```
Comment (jekabs-karklins) [inline apps/user-office-frontend/src/components/SEP/Proposals/SEPProposalsAndAssignmentsTable.tsx:130]: What was the reason for removing the types? Are these types inferred?

**Review by jekabs-karklins — APPROVED** [review]

File: apps/user-office-frontend/src/components/SEP/Proposals/SEPProposalsAndAssignmentsTable.tsx (hunk 3 above)
Comment (Junjiequan) [reply apps/user-office-frontend/src/components/SEP/Proposals/SEPProposalsAndAssignmentsTable.tsx:130]: Yea, I did some cleanup for types here. I assigned the type at the top level, so it won't require types for argument of the render method. You can still check the types when you hover the arg.

File: apps/user-office-frontend-e2e/cypress/integration/proposals.ts (hunk 1 above)
Comment (Junjiequan) [reply apps/user-office-frontend-e2e/cypress/integration/proposals.ts]: It is fixed. thx for pointing out that.

**Pull request merged**